app/services/redis_session_health_checker.rb

```diff
@@ -0,0 +1,30 @@
+module RedisSessionHealthChecker
+  module_function
+
+  # @return [HealthCheckSummary]
+  def check
+    HealthCheckSummary.new(healthy: true, result: health_write_and_read)
+  rescue StandardError => err
+    NewRelic::Agent.notice_error(err)
+    HealthCheckSummary.new(healthy: false, result: err.message)
+  end
+
+  # @api private
+  def health_write_and_read
+    REDIS_POOL.with do |client|
```

Comment:
Not a blocker, but would we want to check all the Redis pools?

Reply:
We could at least add a check for the rate limiting Redis.   Attempts storage is not always enabled so seems like a conditional health check would be needed for that.

Should we combine it all here in one spot and call this `RedisHealthChecker`?

Reply:
I think that makes sense, yeah.

Reply:
I guess they could be separate too, I don't have strong feelings, whatever is easier.